client/src/components/Ratings&Reviews/AddReview.jsx

```diff
@@ -0,0 +1,479 @@
+      photos: [],
+      // urlPhotos: [],
+      productName: '',
+      // error checking
+      ratingError: false,
+      recommendError: false,
+      characteristicsError: false,
+      bodyError: false,
+      nicknameError: false,
+      emailError: false,
+    };
+  }
+
+  componentDidMount() {
+    this.getProductName();
+  }
+
+  // GET product name
+  getProductName() {
+    const { productId } = this.props;
+    $.ajax({
+      url: `/atelier/products/${productId}`,
+      type: 'GET',
+      success: (data) => {
+        this.setState({
+          productName: data.name,
+        });
+      },
+    });
+  }
+
+  // Modal State Updates
+  // Functionality of choosing overall star rating in modal
+  chooseStarRating(e) {
+    const { newReview } = this.state;
+    const overallStar = e.target.id;
+    let num = Number(overallStar[overallStar.length - 1]);
+    let resetStars = 5;
+
+    this.setState({
+      overallStarChosen: Number(overallStar[overallStar.length - 1]),
+    });
+    this.setState({
+      newReview: {
+        ...newReview,
+        rating: num,
+      },
+    });
+    // reset star colors
+    while (resetStars > 0) {
+      const star = $(`#star-${resetStars}`);
+      star.css('color', 'black');
+      resetStars -= 1;
+    }
+    // fill in star colors
+    while (num > 0) {
+      const star = $(`#star-${num}`);
+      star.css('color', 'yellow');
+      num -= 1;
+    }
+  }
+
+  recommendRadio(e) {
+    const { newReview } = this.state;
+    // if yes, true
+    if (e.target.value === 'yes') {
+      this.setState({
+        newReview: {
+          ...newReview,
+          recommend: true,
+        },
+      });
+    }
+    // if no, false
+    if (e.target.value === 'no') {
+      this.setState({
+        newReview: {
+          ...newReview,
+          recommend: false,
+        },
+      });
+    }
+  }
+
+  chooseCharacteristics(e, id) {
+    const { newReview } = this.state;
+    const currentId = id.toString();
+    this.setState({
+      newReview: {
+        ...newReview,
+        characteristics: {
+          ...newReview.characteristics,
+          [currentId]: Number(e.target.value),
+        },
+      },
+    });
+  }
+
+  modalSummary(e) {
+    const { newReview } = this.state;
+    this.setState({
+      newReview: {
+        ...newReview,
+        summary: e.target.value,
+      },
+    });
+  }
+
+  modalBody(e) {
+    const { newReview } = this.state;
+    this.setState({
+      newReview: {
+        ...newReview,
+        body: e.target.value,
+      },
+    });
+  }
+
+  trackWordCount(e) {
+    const { bodyWordCount } = this.state;
+    const newWordCount = e.target.value.length;
+    this.setState({
+      bodyWordCount: newWordCount,
+    });
+    if (bodyWordCount >= 50) {
+      this.setState({
+        bodyWordCountCheck: true,
+      });
+    }
+    if (bodyWordCount < 50) {
+      this.setState({
+        bodyWordCountCheck: false,
+      });
+    }
```

Comment:
Similar idea here: could potentially trim this down by invoking this.setState once and setting bodyWordCountCheck to `bodyWordCountCheck >= 50 ? true : false`